unittests/test_utility.py

```diff
@@ -19,492 +17,547 @@
-
-    def test_strange_syntax(self):
-        text = 'Hello, $(foo`'
-        assert 'Hello, $(foo`' == os_ext.expandvars(text)
-
-        text = 'Hello, `foo)'
-        assert 'Hello, `foo)' == os_ext.expandvars(text)
-
-    def test_expandvars_nocmd(self):
-        os.environ['FOO'] = 'World'
-        text = 'Hello, $FOO'
-        assert 'Hello, World' == os_ext.expandvars(text)
-
-        text = 'Hello, ${FOO}'
-        assert 'Hello, World' == os_ext.expandvars(text)
-        del os.environ['FOO']
-
-
-class TestCopyTree(unittest.TestCase):
-    def setUp(self):
-        # Create a test directory structure
-        #
-        # prefix/
-        #   bar/
-        #     bar.txt
-        #     foo.txt
-        #     foobar.txt
-        #   foo/
-        #     bar.txt
-        #   bar.txt
-        #   foo.txt
-        #
-        self.prefix = os.path.abspath(tempfile.mkdtemp())
-        self.target = os.path.abspath(tempfile.mkdtemp())
-        os.makedirs(os.path.join(self.prefix, 'bar'), exist_ok=True)
-        os.makedirs(os.path.join(self.prefix, 'foo'), exist_ok=True)
-        open(os.path.join(self.prefix, 'bar', 'bar.txt'), 'w').close()
-        open(os.path.join(self.prefix, 'bar', 'foo.txt'), 'w').close()
-        open(os.path.join(self.prefix, 'bar', 'foobar.txt'), 'w').close()
-        open(os.path.join(self.prefix, 'foo', 'bar.txt'), 'w').close()
-        open(os.path.join(self.prefix, 'bar.txt'), 'w').close()
-        open(os.path.join(self.prefix, 'foo.txt'), 'w').close()
-
-        # Create also a subdirectory in target, so as to check the recursion
-        os.makedirs(os.path.join(self.target, 'foo'), exist_ok=True)
-
-    def verify_target_directory(self, file_links=[]):
+    return _rmtree
+
+
+def test_rmtree(rmtree):
+    rmtree()
+
+
+def test_rmtree_onerror(rmtree):
+    rmtree(onerror=lambda *args: None)
+
+
+def test_rmtree_error(tmp_path):
+    # Try to remove an inexistent directory
+    testdir = tmp_path / 'tmp'
+    testdir.mkdir()
+    os.rmdir(str(testdir))
+    with pytest.raises(OSError):
+        os_ext.rmtree(testdir)
+
+
+def test_inpath():
+    assert os_ext.inpath('/foo/bin', '/bin:/foo/bin:/usr/bin')
+    assert not os_ext.inpath('/foo/bin', '/bin:/usr/local/bin')
+
+
+@pytest.fixture
+def prefix(tmp_path):
+    prefix = tmp_path / 'prefix'
+    return prefix
+
+
+@pytest.fixture
+def testdirs(prefix):
+    # Create a temporary directory structure
+    # foo/
+    #   bar/
+    #     boo/
+    #   goo/
+    # loo/
+    #   bar/
+    (prefix / 'foo' / 'bar' / 'boo').mkdir(parents=True)
+    (prefix / 'foo' / 'goo').mkdir()
+    (prefix / 'loo' / 'bar').mkdir(parents=True)
+
+
+def test_subdirs(prefix, testdirs):
+    # Try to fool the algorithm by adding normal files
+    prefix_name = str(prefix)
+    open(os.path.join(prefix_name, 'foo', 'bar', 'file.txt'), 'w').close()
+    open(os.path.join(prefix_name, 'loo', 'file.txt'), 'w').close()
+
+    expected_subdirs = {prefix_name,
```

Comment:
```suggestion
    expected_subdirs = {prefix.name,
```

And replace that everywhere.